### ISSUE TITLE
[Server Request] Feed the Beast - Minecraft Forge

### ISSUE THREAD

**Issue by xopez** [GameServerManagers/LinuxGSM#2299]
## What game server would you like adding?
Feed the Beast provides many Modpacks for minecraft with forge. Would be nice to have it in lgsm. You can read here more about it: https://ftb.gamepedia.com/FTB_Wiki
List of possible Packs https://www.feed-the-beast.com/modpacks

## Can you link to any tutorials or guides?

Provide any useful documentation or resources that might help.

## Is the server on Steam?

* [ ] Yes
* [x] No

## Is the server supported on Linux?

We only support Linux servers and do not support Wine.
since minecraft is java based its fully compatible.
* [x] Yes
* [ ] No

**Comment by Aidoboy** [comment]
Why not just CurseForge modpacks: https://minecraft.curseforge.com/modpacks
This could also help: https://github.com/Vazkii/CMPDL

**Comment by xopez** [comment]
Good point. I don't how much the packs differ because they are named close at the same to feet the beast

**Comment by Aidoboy** [comment]
The CurseForge list includes the FTB list, Amazon owns Twitch owns Curse owns FTB.

**Comment by KILLERMANTV** [comment]
You don't need a extra game server for that, just download minecraft server and replace all the vanilla files with modpack files, maybe you can also assign more ram for a modpack.

**Comment by Aidoboy** [comment]
That tutorial isn't for LGSM, which has some particular things you have to do to set up a server. It would be really nice to have the process automated 

It also doesn't use the CurseForge server download, for some reason it's just copying the single player files? There's sometimes a difference.

**Comment by Chakratos** [comment]
I would like to push this request as i have the same one.
Sure you can do it manually but updating everything manually is a pain. It would be great if the update command would get the latest updates from the FTB Modpack itself.

**Comment by dgibbs64** [comment]
Minecraft is next on my list after rust for review. Specifically, the best way to get all these variants installs automatically.

**Comment by UndarkAido** [comment]
It's not done yet, but here's the guide for the new FTB server installers: https://faq.ftb.world/books/ftb-app/page/downloading-server-installerfiles

I'm having some issues with the Omnia installer so I'm going to see if I can get that worked out.

**Comment by NateZeNarwhal** [comment]
THIS PLEASE!!! I have been struggling all day to get a modded Minecraft server to work with GSM. 

One note I do have: Use CurseForge. This includes modpacks from Tekkit and FTB. It's the most widely used modpack service atm. [https://www.curseforge.com/minecraft/modpacks](url)

**Comment by UndarkAido** [comment]
FTB is leaving the Curse / Twitch launcher @NateZeNarwhal https://forum.feed-the-beast.com/threads/ftb-update-new-launcher-ftb-omnia-released.302085/

**Comment by Chakratos** [comment]
What i did to get it running was downloading the Linux server tool from modpacks.ch (Its not modpack specific as you can search for packs with it)
https://api.modpacks.ch/public/modpack/35/37/server/linux

Install a normal Minecraft server with LinuxGSM -> Insert the file into Serverfiles and execute it.
![grafik](https://user-images.githubusercontent.com/8162256/82552456-6dd50a00-9b62-11ea-9abb-783547f6465e.png)

After choosing your modpack it will install automatically.
Then you just have to edit your:
lgsm/config-lgsm/mcserver/common.cfg

and insert:
executable="java -server -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -Xmx8192M -Xms6124M -jar ${serverfiles}/forge-1.12.2-14.23.5.2846-universal.jar"

(You can change the ram here if needed. Additionaly you might have to rename the forge.jar file.)

Now you can use your Modpack server with linuxGSM

**Comment by James2024** [comment]
> 
> 
> What i did to get it running was downloading the Linux server tool from modpacks.ch (Its not modpack specific as you can search for packs with it)
> https://api.modpacks.ch/public/modpack/35/37/server/linux
> 
> It seems like every mayor Modpack is available there.
> 
> Install a normal Minecraft server with LinuxGSM -> Insert the file into Serverfiles and execute it.
> ![grafik](https://user-images.githubusercontent.com/8162256/82552456-6dd50a00-9b62-11ea-9abb-783547f6465e.png)
> 
> After choosing your modpack it will install automatically.
> Then you just have to edit your:
> lgsm/config-lgsm/mcserver/common.cfg
> 
> and insert:
> executable="java -server -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -Xmx8192M -Xms6124M -jar ${serverfiles}/forge-1.12.2-14.23.5.2846-universal.jar"
> 
> (You can change the ram here if needed. Additionaly you might have to rename the forge.jar file.)
> 
> Now you can use your Modpack server with linuxGSM
> 
> PS: If starting your Minecraft Server with linuxGSM fails, you probably have JAVA9 installed. You need to use Java8 for the Modded server to run as 9 is NOT supportet.

So what I have been doing is launching linuxGSM getting the files then just replacing all files in /serverfiles with files I generated on my PC for my modpack.

But it doesnt start I just get "[ FAIL ] Starting mcserver: Unable to start server" message. Is this because I need Java 8? and if so how do I downgrade to java 8 and use it instead of java 9.

**Comment by KILLERMANTV** [comment]
> But it doesnt start I just get "[ FAIL ] Starting mcserver: Unable to start server" message. Is this because I need Java 8? and if so how do I downgrade to java 8 and use it instead of java 9.

To install java 8 run this:
`sudo apt install openjdk-8-jdk openjdk-8-jre`

To select it for system to use type this:
`sudo update-alternatives --config java`

and type the number of line that has java-8 in it.

**Comment by dgibbs64** [comment]
Minecraft is currently being reqired with Plans to support Forge. However, there are no plans to support specific modpacks as there are hundreds

**Comment by UndarkAido** [comment]
> Minecraft is currently being reqired with Plans to support Forge. However, there are no plans to support specific modpacks as there are hundreds

You wouldn't have to do anything to support individual Modpacks: I think CurseForge would require some kind of agreement with Overwolf but the FTB packs can all be downloaded with that standard script.

If anything support for Modpacks `zip`s would be nice

**Comment by hisakiyo** [comment]
> > What i did to get it running was downloading the Linux server tool from modpacks.ch (Its not modpack specific as you can search for packs with it)
> > https://api.modpacks.ch/public/modpack/35/37/server/linux
> > It seems like every mayor Modpack is available there.
> > Install a normal Minecraft server with LinuxGSM -> Insert the file into Serverfiles and execute it.
> > ![grafik](https://user-images.githubusercontent.com/8162256/82552456-6dd50a00-9b62-11ea-9abb-783547f6465e.png)
> > After choosing your modpack it will install automatically.
> > Then you just have to edit your:
> > lgsm/config-lgsm/mcserver/common.cfg
> > and insert:
> > executable="java -server -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -Xmx8192M -Xms6124M -jar ${serverfiles}/forge-1.12.2-14.23.5.2846-universal.jar"
> > (You can change the ram here if needed. Additionaly you might have to rename the forge.jar file.)
> > Now you can use your Modpack server with linuxGSM
> > PS: If starting your Minecraft Server with linuxGSM fails, you probably have JAVA9 installed. You need to use Java8 for the Modded server to run as 9 is NOT supportet.
> 
> So what I have been doing is launching linuxGSM getting the files then just replacing all files in /serverfiles with files I generated on my PC for my modpack.
> 
> But it doesnt start I just get "[ FAIL ] Starting mcserver: Unable to start server" message. Is this because I need Java 8? and if so how do I downgrade to java 8 and use it instead of java 9.

You can try to add to common.cfg
```
preexecutable=""
```
